binding/python/test_leopard.py

```diff
@@ -9,100 +9,171 @@
-            self.assertAlmostEqual(word.start_sec, ref_word.start_sec, delta=0.01)
-            self.assertAlmostEqual(word.end_sec, ref_word.end_sec, delta=0.01)
-            self.assertAlmostEqual(word.confidence, ref_word.confidence, delta=0.1)
-
-    @parameterized.expand(TEST_PARAMS)
-    def test_process(self, enable_automatic_punctuation: bool, ref_transcript: str):
+        cls._access_key = sys.argv[1]
+        cls._audio_directory = os.path.join('..', '..', 'resources', 'audio_samples')
+
+    def _validate_metadata(self, words: Sequence[Leopard.Word], transcript: str, audio_length: int):
+        norm_transcript = transcript.upper()
+        for i in range(len(words)):
+            self.assertTrue(words[i].word.upper() in norm_transcript)
+            self.assertGreater(words[i].start_sec, 0)
+            self.assertLessEqual(words[i].start_sec, words[i].end_sec)
+            if i < len(words) - 1:
+                self.assertLessEqual(words[i].end_sec, words[i + 1].start_sec)
+            else:
+                self.assertLessEqual(words[i].end_sec, audio_length)
+            self.assertTrue(0 <= words[i].confidence <= 1)
+
+    def test_invalid_access_key(self):
+        with self.assertRaises(LeopardInvalidArgumentError):
+            Leopard(
+                access_key='invalid',
+                model_path=default_model_path('../../'),
+                library_path=default_library_path('../../'))
+
+    def test_invalid_model_path(self):
+        with self.assertRaises(LeopardIOError):
+            Leopard(
+                access_key=self._access_key,
+                model_path='invalid',
+                library_path=default_library_path('../../'))
+
+    def test_invalid_library_path(self):
+        with self.assertRaises(LeopardIOError):
+            Leopard(
+                access_key=self._access_key,
+                model_path=default_model_path('../../'),
+                library_path='invalid')
+
+    def test_version(self):
+        o = Leopard(
+            access_key=self._access_key,
+            model_path=default_model_path('../../'),
+            library_path=default_library_path('../../'))
+        self.assertIsInstance(o.version, str)
+        self.assertGreater(len(o.version), 0)
+
+    @parameterized.expand(parameters)
+    def test_process(
+            self,
+            language: str,
+            audio_file: str,
+            expected_transcript: str,
+            punctuations: List[str],
+            error_rate: float):
         o = None
 
         try:
-            o = self._create_leopard(enable_automatic_punctuation=enable_automatic_punctuation)
-            transcript, words = o.process(self.pcm)
-            self._check_result(transcript=transcript, ref_transcript=ref_transcript, words=words)
+            o = Leopard(
+                access_key=self._access_key,
+                model_path=get_model_path_by_language(relative='../../', language=language),
```

Comment:
In other repos (for `node` I think) we've been getting rid of `relative` as an argument to all these util functions and just putting it inside the function itself since it's just set to `../../` everywhere anyways.